lib/Sema/BuilderTransform.cpp

```diff
@@ -459,6 +459,33 @@ class BuilderClosureVisitor
                                  /*isTopLevel=*/true);
   }
 
+  /// If there is a #available in the condition, the 'then' will be
+  /// wrapped in a call to buildLimitedAvailability(_:), if available.
+  /// If this is instead a #unavailability condition, the 'else' will
+  /// be wrapped instead.
+  Expr *wrapInLimitedAvailabilityIfNeeded(IfStmt *ifStmt, Expr *arg, 
+                                          bool isUnavailability = false) {
+    auto availabilityCond = findAvailabilityCondition(ifStmt->getCond());
+    bool supportsAvailable = builderSupports(ctx.Id_buildLimitedAvailability);
+
+    auto apiAvailability = availabilityCond->getAvailability();
+    if (availabilityCond && 
+        apiAvailability->isUnavailability() == isUnavailability &&
+        supportsAvailable) {
+      SourceLoc endLoc;
+      if (isUnavailability) {
+        endLoc = ifStmt->getEndLoc();
+      } else {
+        endLoc = ifStmt->getThenStmt()->getEndLoc();
+      }
+      return buildCallIfWanted(
+          ifStmt->getThenStmt()->getEndLoc(), ctx.Id_buildLimitedAvailability,
```

Comment:
This seems like a bug; should it be the `endLoc` variable instead of `ifStmt->getThenStmt()->getEndLoc()`?

Reply:
I think in fact that the whole implementation of the builder in this PR was confusing. I didn't know what I was changing in the beginning and I see now that it was over-complicated. I fixed that up and cleaned it up, though I'm not sure exactly where this loc is using to write a test for it, it seems to be only for diagnostics but they appear to the working fine even with the bug.